src/diffusers/models/unet_2d_condition.py

```diff
@@ -937,8 +949,8 @@ def forward(
                 encoder_attention_mask=encoder_attention_mask,
             )
 
-        if mid_block_additional_residual is not None:
-            sample = sample + mid_block_additional_residual
+        if is_controlnet:
+            sample += mid_block_additional_residual
```

Comment:
Wasn't there a problem with this: https://github.com/huggingface/diffusers/pull/2586/files would prefer to leave as is

Reply:
ah yeah, just didn't see when rebasing